docs/design/storage-node-deployment-windows-home.md

```diff
@@ -0,0 +1,146 @@
+# Storage Node Automatic Updates and Installation for Windows Home
+
+## Overview
+
+As more Storage Node Operators join the network we must ensure these nodes have a mechanism to automatically keep their nodes software up to date.
+If a Storage Node Operator fails to keep their node up to date with the minimum version required by satellites they will no longer be selected for upload or download requests.
+
+Currently we are using Docker for automatic updates which means we require Docker to be installed.
+Docker is being used for many other things as well, so we need to cover these cases as well.
+
+## Goals
+
+Docker is being used for:
+
+* Automatic Updates (with watchtower)
+* Restarting on crash
+* Logging (kind of)
+
+To deploy the automatic updates we need to handle these cases.
+
+Docker is not supported on windows home so we will ensure an automatic update system is built into the Storage nodes. We also need to ensure that it plays nicely with all the common anti-viruses and firewalls.
+
+## Services
+* Installer (msi)
+    * Installs automatic updater binary and error gui application
+    * Sets up automatic updater Service with sufficient privileges.
+* Automatic Updater (binary)
+    * Downloads, Sets up, and runs the storage node.
+    * If storage node has not been setup we don't want to try to run the storage node.
+    * Send error reports to satellite.
+    * Writes errors to log file
+* Storage Node (binary)
+    * shares drive with satellite network.
+* Error gui application
```

Comment:
Why is this a separate application?

Reply:
The purpose of it is for notifying the user of any errors. The automatic updater will run as a service so that it may continue running when not logged in. The error application however only needs to be running while a user is logged in.

Reply:
Yes, but as a result it would need to communicate with the storage node binary. I think there might be a way to handle notifications without having to make a separate binary.

e.g. https://docs.microsoft.com/en-us/windows/uwp/design/shell/tiles-and-notifications/toast-desktop-apps